compiler/src/dotty/tools/dotc/transform/ReifyQuotes.scala

```diff
@@ -199,8 +202,29 @@ class ReifyQuotes extends MacroTransform {
     }
 
     private def pickledQuote(body: Tree, splices: List[Tree], originalTp: Type, isType: Boolean)(implicit ctx: Context) = {
-      def pickleAsValue[T](value: T) =
-        ref(defn.Unpickler_liftedExpr).appliedToType(originalTp.widen).appliedTo(Literal(Constant(value)))
+
+      def liftedValue[T](value: T, name: TermName, qctx: Tree) =
+        ref(defn.LiftableModule).select(name).select("toExpr".toTermName).appliedTo(Literal(Constant(value))).appliedTo(qctx)
+
+      def pickleAsValue[T](value: T) = {
+        val qctx = ctx.typer.inferImplicitArg(defn.QuoteContextType, body.span)
+        if (qctx.tpe.isInstanceOf[SearchFailureType])
+          ctx.error(ctx.typer.missingArgMsg(qctx, defn.QuoteContextType, ""), ctx.source.atSpan(body.span))
+        value match {
+          case null => ref(defn.QuotedExprModule).select("nullExpr".toTermName).appliedTo(qctx)
+          case _: Unit => ref(defn.QuotedExprModule).select("unitExpr".toTermName).appliedTo(qctx)
+          case _: Boolean => liftedValue(value, "Liftable_Boolean_delegate".toTermName, qctx)
+          case _: Byte => liftedValue(value, "Liftable_Byte_delegate".toTermName, qctx)
+          case _: Short => liftedValue(value, "Liftable_Short_delegate".toTermName, qctx)
+          case _: Int => liftedValue(value, "Liftable_Int_delegate".toTermName, qctx)
+          case _: Long => liftedValue(value, "Liftable_Long_delegate".toTermName, qctx)
+          case _: Float => liftedValue(value, "Liftable_Float_delegate".toTermName, qctx)
+          case _: Double => liftedValue(value, "Liftable_Double_delegate".toTermName, qctx)
+          case _: Char => liftedValue(value, "Liftable_Char_delegate".toTermName, qctx)
+          case _: String => liftedValue(value, "Liftable_String_delegate".toTermName, qctx)
+        }
+      }
+
```

Comment:
We still need special compiler support for lifting primitives? Or it's for backward-compatibility.

Reply:
We do not need it anymore.